packages/medusa/package.json

```diff
@@ -27,15 +28,22 @@
     "jest": "^25.5.2",
     "medusa-interfaces": "^1.3.3",
     "nodemon": "^2.0.1",
+    "openapi-typescript-codegen": "https://github.com/patrick-medusajs/openapi-typescript-codegen#8c7bf94be290a9b40d31b55fd2ef4421d56f31d7",
     "sqlite3": "^5.0.2",
     "supertest": "^4.0.2",
+    "swagger-inline": "^6.1.0",
     "ts-jest": "^25.5.1",
+    "ts-node": "^10.9.1",
     "typescript": "^4.4.4"
   },
   "scripts": {
     "watch": "tsc --build --watch",
     "prepare": "cross-env NODE_ENV=production yarn run build",
-    "build": "tsc --build",
+    "build": "tsc --build && yarn run build:oas && yarn build:client",
```

Comment:
Agreed, this is not ideal and I have a task to further explore the build and publishing strategy. 

That being said, I do like having immediate feedback if the OAS is in a broken state due to a code change. This would become critical if we increase code-coupling with the OAS and if we increase our dependency on the OAS being accurate and in a shippable state.

Reply:
Reworked the DevX to remove OAS from build pipeline. Introduced a `--onlyValidate` flag to `build:oas` to be used by the oas-test.yml CI workflow. Should be enough safety and not intrusive during local development.